kernel/src/hil/usb_hid.rs

```diff
@@ -0,0 +1,67 @@
+//! Interface for USB HID (Human Interface Device) class
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    /// Called when a packet is received.
+    /// This will return the buffer passed into `receive_buffer()` as well as
+    /// the endpoint where the data was received.
+    fn packet_received(&'a self, buffer: &'static mut [u8; 64], len: usize, endpoint: usize);
+
+    /// Called when a packet has been finished transmitting.
+    /// This will return the buffer passed into `send_buffer()` as well as
+    /// the endpoint where the data was sent.
+    fn packet_transmitted(
+        &'a self,
+        result: Result<(), ReturnCode>,
+        buffer: &'static mut [u8; 64],
+        len: usize,
+        endpoint: usize,
+    );
+
+    /// Called when checking if we can receive any more data.
+    /// Should return true if we are ready to receive.
+    fn can_receive(&'a self) -> bool;
```

Comment:
I have changed this to:

```Rust
    /// Called when checking if we can start a new receive operation.
    /// Should return true if we are ready to receive and not currently
    /// in the process of receiving anything. That is if we are currently
    /// idle.
    /// If there is an outstanding call to receive, a callback already
    /// waiting to be called then this will return false.
```
to be more clear